R/configuration-plan.R

```diff
@@ -132,9 +132,18 @@ ConfigurationPlan <- R6::R6Class(
     getObservedDataPath = function(id) {
       validateIsIncluded(id, private$.observedDataSets$id, groupName = "'id' variable of observedDataSets")
       selectedId <- private$.observedDataSets$id %in% id
-      # Minimal example includes a case where multiple same ids were used
-      # What should be done in such cases ? (Currently, firt value is used)
-      return(file.path(self$referenceFolder, private$.observedDataSets$path[selectedId])[1])
+      # In case of duplicate observed data, use first
+      return(utils::head(file.path(self$referenceFolder, private$.observedDataSets$path[selectedId]), 1))
+    },
+    
+    #' @description Get molecular weight of observed data corresponding to a specific observedDataSet Id
+    #' @param id observedDataSet identifier
+    #' @return The observed data file path corresponding to the id in the configuration plan field `observedDataSet`
+    getMolWeightForObservedData = function(id) {
+      validateIsIncluded(id, private$.observedDataSets$id, groupName = "'id' variable of observedDataSets")
+      selectedId <- private$.observedDataSets$id %in% id
+      # In case of duplicate observed data, use first
+      return(utils::head(private$.observedDataSets$molWeight[selectedId], 1))
     },

```

Comment:
don't understand the implementation

Reply:
`ConfigurationPlan` objects  include a field named `observedDataSets` that translated the same field in the json file of the configuration plan. This field is a data.frame with `id`, `path`, `type` and now `molWeight` variables
So to get a path or a specific observed dataset `id`, the method 1) checks that `id` is included in the config plan; 2) select the row corresponding to the id (in observed data set, there were cases of duplicate ids and paths in which case they are identical but only one value needs to be returned); 3) return the `path` or `molWeight` for that selected row.